### ISSUE TITLE
Use nilearn's resample_img for subsampling

### ISSUE THREAD

**Issue by tsalo** [ME-ICA/mapca#31]
How do you all feel about using [`nilearn.image.resample_img()`](https://nilearn.github.io/modules/generated/nilearn.image.resample_img.html#nilearn.image.resample_img) for `utils.subsampling()`? I am working on #29, and I think it would be easier have a function that downsamples and returns images instead of arrays.

**Comment by tsalo** [comment]
Actually, downsampling and subsampling may be different tasks. We can still make the subsampling function work on images though.

**Comment by eurunuela** [comment]
> How do you all feel about using [`nilearn.image.resample_img()`](https://nilearn.github.io/modules/generated/nilearn.image.resample_img.html#nilearn.image.resample_img) for `utils.subsampling()`? I am working on #29, and I think it would be easier have a function that downsamples and returns images instead of arrays.

I don't think `nilearn.image.resample_img()` is what we're looking for. 

> Actually, downsampling and subsampling may be different tasks. We can still make the subsampling function work on images though.

I don't see why we would need to do this. The subsampling happens inside the fitting function, where images are already in matrix format. Is there any other case where we would want to subsample images?

**Comment by tsalo** [comment]
Agreed that `resample_img` is not the right tool. However, I will say that having `subsampling` operate on images will make it easier to stick with `nilearn`'s masking and unmasking functions internally. At that point, though, the results will be the same and the changes will be buried deep in the code, so it's not the big change that using `resample_img` would be.
